pkg/serverconfig/validate.go

```diff
@@ -127,6 +132,39 @@ func validateProjectAccessClusterRolesJSON(value string) ([]string, error) {
 	return projectAccessOptions, nil
 }
 
+func validatePerspectives(value string) ([]Perspective, error) {
+	if value == "" {
+		return nil, nil
+	}
+	var perspectives []Perspective
+
+	decoder := json.NewDecoder(strings.NewReader(value))
+	decoder.DisallowUnknownFields()
+	if err := decoder.Decode(&perspectives); err != nil {
+		return nil, err
+	}
+
+	for perspectiveIndex, perspective := range perspectives {
+		if perspective.ID == "" {
+			return perspectives, fmt.Errorf("Perspective at index %d must have id property.", perspectiveIndex)
+		}
+		if perspective.ID != "" && reflect.DeepEqual(perspective.Visibility, PerspectiveVisibility{}) {
```

Comment:
ID is already checked, so we can focus on visibility here and keep it simple:

```suggestion
		if perspective.Visibility.State == "" {
```